webpack/assets/javascripts/react_app/common/TranslateMapper/index.js

```diff
@@ -0,0 +1,34 @@
+import forceSingleton from '../forceSingleton';
+
+/**
+ * A mapper for runtime translations
+ */
+
+class TranslateMapper {
+  static registery = forceSingleton('translateMapper', () => ({}));
+
+  /**
+   * Register string to the translate mapper
+   * @param  {string} id a uniqe id
+   * @param  {Object} translatedStrings  key-value pairs object of translated strings
+   */
+
+  static register = translatedStrings => {
+    for (const property in translatedStrings) {
+      this.registery[property] = translatedStrings[property];
```

Comment:
Do we care about collisions?

Reply:
we can use collisions for override strings when needed